src/features/client/components/RestaurantList/index.jsx

```diff
@@ -1,23 +1,70 @@
-import { Box, Flex, Heading } from '@chakra-ui/react';
+import { Box, Center, Flex, Heading, Spinner, Text } from '@chakra-ui/react';
+import { useInfiniteQuery } from 'react-query';
 
-import { restaurants } from '../../mocks/restaurantsMock';
 import RestaurantListCard from '../RestaurantListCard';
+import merchantService from '../../../../services/merchantService';
+import { useEffect, useState } from 'react';
+import { useInView } from 'react-intersection-observer';
 
-function RestaurantList({ isFeatured, selectedCategory }) {
-  let restaurantsList = isFeatured
-    ? restaurants.filter((item) => item.featured)
-    : restaurants;
+function RestaurantList({ selectedCategory, currentMerchantType }) {
+  const { ref, inView } = useInView();
+  const [queryURL, setQueryURL] = useState(null);
 
-  if (selectedCategory && selectedCategory !== 'All') {
-    restaurantsList = restaurantsList.filter((restaurant) =>
-      restaurant.restaurantTags.includes(selectedCategory)
+  const {
+    data: fetchedRestaurantData,
+    isFetching,
+    fetchNextPage,
+    isFetchingNextPage,
+    hasNextPage,
+  } = useInfiniteQuery(
+    ['merchants', `filter_equals_type=${queryURL || 'restaurant'}`],
+    merchantService.getMerchantList,
+    {
+      getNextPageParam: (lastPage) =>
+        lastPage.links.next?.split('merchants?')[1] ?? undefined,
+    }
+  );
+
+  const [restaurantsList, setRestaurantsList] = useState(
+    fetchedRestaurantData?.pages.reduce((acc, cur) => {
+      return [...acc, ...cur.data];
+    }, []) || []
+  );
+  const latestFetchedRestaurantList =
+    fetchedRestaurantData?.pages[fetchedRestaurantData?.pages?.length - 1].data;
+
+  useEffect(() => {
+    if (!isFetching && !isFetchingNextPage && hasNextPage) fetchNextPage();
+  }, [inView]);
+
+  useEffect(() => {
+    setRestaurantsList([]);
+    setQueryURL(
+      selectedCategory
+        ? `${currentMerchantType}&filter_equals_categoryId=${selectedCategory?.id}`
+        : currentMerchantType
     );
-  }
+  }, [selectedCategory, currentMerchantType]);
+
+  useEffect(() => {
+    const existCheck = restaurantsList?.filter(
```

Comment:
boolean flag should formed with `is` `does` `has`

in this case rename to `restaurantDoesExist`